### PR TITLE
Import a new version of pex (v1.1.1)

### PR DESCRIPTION
**Pull request by mikekap**
The new release properly does site-packages isolation, so pexs will no longer pull in locally installed packages on newer distros. You can use PEX_INHERIT_PATH=1 to revert to the (almost) the old behavior.

Other niceties include:
 - access_zipped_assets() [just a pkg_resources crawl; has nothing to do with zips]
 - perf improvements for extracting prebuilt packages
 - ~/.pexrc, /etc/pexrc for local/global pex settings (like PEX_ROOT)
 - a bunch of new env variables: PEX_SCRIPT/PEX_PATH/PEX_PYTHON. I personally don't recommend using PEX_PATH (allows combining pexs) :(.

### CODE REVIEW

**Comment by mikekap** [comment]
Hmm I think that test failure is unrelated: `/tmp/buck-d89336aa-eab8-4dea-b8b6-a5468497e748/test/com/facebook/buck/python/__java_test_python_tmp__/junit6721994106692138210/buck-out/gen/bin-with-native-libs#link-tree/libfoo.so` seems to be coming from an inplace python executable.

**Comment by sdwilsh** [comment]
That Travis failure looks like it may very well be related.

File: third-party/py/pex/README.facebook [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Project URL: 
+Git repo to clone: https://github.com/pantsbuild/pex.git
+Git revision: 1e70fddafd480311e717e58dbf9466cf40003137
+License: Apache 2.0
+Local modifications:
+ - Only imported the pex source (no tests/docs).
+ - Disabled the part of the twitter PEX code that auto-detects the setuptools package from the
+   system, as this got more problematic to fake in our own `pkg_resources` module.
+ - Fixed usage of os.link (does not exist on windows).
```
Comment (sdwilsh) [inline third-party/py/pex/README.facebook]: We really should upstream some of this code one day...

**Comment by sdwilsh** [comment]
@facebook-github-bot import

File: src/com/facebook/buck/python/make_pex.py [hunk 2]
```diff
@@ -16,23 +16,17 @@
 # setuptools at runtime.  Also, locate the `pkg_resources` modules
 # via our local setuptools import.
 if not zipfile.is_zipfile(sys.argv[0]):
-    # Remove twitter.common.python from the import path - it may be eagerly
-    # loaded as part of site-packages.
-    sys.modules.pop('twitter', None)
-    sys.modules.pop('twitter.common', None)
-    sys.modules.pop('twitter.common.python', None)
```
Comment (sdwilsh) [inline src/com/facebook/buck/python/make_pex.py]: Do we not need to worry about removing this stuff if we aren't a zipfile?

**Comment by sdwilsh** [comment]
This is a little difficult to review and make sure it's safe.  Could you do a plain import to the new location, apply the patches we need to fix things for, and then do the cutover?  That will be easier to review (with two of them being essentially stamps), but it'll also be easier to update in the future because the changes that need to be made to the pex code could be unapplied and then reapplied with a cherry-pick.

**Comment by mikekap** [comment]
Sure no problem - would you prefer them in multiple PRs or just multiple commits?

**Comment by sdwilsh** [comment]
Multiple PRs would work best because our import flow squashes all of your commits.  I'm trying to make sure this is maintainable and easier to update in the future so we don't have this big pain again.

**Comment by mikekap** [comment]
Done. You can see the 5 commits @ https://github.com/mikekap/buck/commits/unsquashed_pex

**Comment by sdwilsh** [comment]
@facebook-github-bot shipit

**Comment by sdwilsh** [comment]
Fantastic.  Hopefully we can get all those merged in today before I have to leave on a flight (I really wish GitHub made it easier to do stacked pull requests).

**Comment by sdwilsh** [comment]
Sorry this took so long to get in.  We are having some infra issues internally that prevented it from automatically happening.

**Pull request closed without merging**